File: Exercises/JavaScript/JavaScript_Basics/Variable_Scope/ex10.js
// What will the following code log to the console and why? Don't run it until you have tried to answer.

const a = {
  firstName: 'John',
  lastName: 'Doe'
};

function myFunction() {
  a.firstName = 'Jane';
}

myFunction();

console.log(a);

// it will print 'Jane' because you can re-assign properties of an object a constant points to, but not the object a constant ponits to
// const values can be mutated, but the values that are pointing to can't

File: Exercises/Python/Python_Basics/6_Variable-Scope/ex10.py
'''
What will the following code do and why? Don't run it until you have tried to answer.
'''

b = [1, 2, 3]

def my_function():
    b[0] = 10

my_function()
print(b)


'''
it will print [10, 2, 3] - 'b' in the function block references the block above it because lists are mutable and can be changed
therefore it is not a re-assignment of a variable, it is a mutation of a variable
'''
